File: Serverless-CRUD/Create.cs
using System.Text.Json;
using Amazon.Lambda.APIGatewayEvents;
using Amazon.Lambda.Core;
using MySqlConnector;
using System.Threading.Tasks;
using System;
using System.Collections.Generic;

namespace ServerlessCrud
{
    public class Create
    {
        public async Task<APIGatewayProxyResponse> FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context)
        {
            try
            {
                if (string.IsNullOrWhiteSpace(request?.Body))
                {
                    return JsonResponse(400, new { message = "Request body is required." });
                }

                var input = JsonSerializer.Deserialize<CreateTaskDto>(request.Body, new JsonSerializerOptions
                {
                    PropertyNameCaseInsensitive = true
                });

                if (input == null || string.IsNullOrWhiteSpace(input.Title))
                {
                    return JsonResponse(400, new { message = "Title is required." });
                }

                await using var connection = CreateConnectionFromEnv();
                await connection.OpenAsync();

                const string sql = @"INSERT INTO Tasks (Title, Description, Status)
                                     VALUES (@title, @description, @status);
                                     SELECT LAST_INSERT_ID();";

                await using var cmd = new MySqlCommand(sql, connection);
                cmd.Parameters.AddWithValue("@title", input.Title);
                cmd.Parameters.AddWithValue("@description", input.Description ?? string.Empty);
                cmd.Parameters.AddWithValue("@status", string.IsNullOrWhiteSpace(input.Status) ? "To Do" : input.Status);

                var idObj = await cmd.ExecuteScalarAsync();
                var id = Convert.ToInt32(idObj);

                var created = new TaskEntity
                {
                    Id = id,
                    Title = input.Title,
                    Description = input.Description ?? string.Empty,
                    Status = string.IsNullOrWhiteSpace(input.Status) ? "To Do" : input.Status
                };

                return JsonResponse(201, created);
            }
            catch (Exception ex)
            {
                context.Logger.LogError(ex.ToString());
                return JsonResponse(500, new { message = "Internal Server Error" });
            }
        }

        private static MySqlConnection CreateConnectionFromEnv()
        {
            var host = Environment.GetEnvironmentVariable("DB_HOST");
            var port = Environment.GetEnvironmentVariable("DB_PORT") ?? "3306";
            var user = Environment.GetEnvironmentVariable("DB_USER");
            var password = Environment.GetEnvironmentVariable("DB_PASSWORD");
            var database = Environment.GetEnvironmentVariable("DB_NAME");

            var cs = $"Server={host};Port={port};Database={database};User ID={user};Password={password};SslMode=Preferred;";
            return new MySqlConnection(cs);
        }

        private static APIGatewayProxyResponse JsonResponse(int statusCode, object payload)
        {
            return new APIGatewayProxyResponse
            {
                StatusCode = statusCode,
                Body = JsonSerializer.Serialize(payload),
                Headers = new Dictionary<string, string>
                {
                    { "Content-Type", "application/json" },
                    { "Access-Control-Allow-Origin", "*" },
                    { "Access-Control-Allow-Methods", "GET,POST,PUT,DELETE,OPTIONS" },
                    { "Access-Control-Allow-Headers", "Content-Type,Authorization" }
                }
            };
        }
    }

    internal class CreateTaskDto
    {
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Status { get; set; }
    }

    internal class TaskEntity
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Status { get; set; } = "To Do";
    }
}

File: serverless/create.py
import json
import os
import base64
import pymysql


def _headers():
    return {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
        "Access-Control-Allow-Headers": "Content-Type,Authorization",
    }


def _conn():
    return pymysql.connect(
        host=os.environ["DB_HOST"],
        port=int(os.environ.get("DB_PORT", "3306")),
        user=os.environ["DB_USER"],
        password=os.environ["DB_PASSWORD"],
        database=os.environ["DB_NAME"],
        charset="utf8mb4",
        cursorclass=pymysql.cursors.DictCursor,
        autocommit=True,
    )


def _parse_body(event):
    body = event.get("body")
    if not body:
        return {}
    if event.get("isBase64Encoded"):
        body = base64.b64decode(body).decode("utf-8")
    try:
        return json.loads(body)
    except Exception:
        return {}


def lambda_handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return {"statusCode": 204, "headers": _headers(), "body": ""}

    try:
        data = _parse_body(event)
        title = (data.get("Title") or data.get("title") or "").strip()
        if not title:
            return {
                "statusCode": 400,
                "headers": _headers(),
                "body": json.dumps({"message": "Title is required."}),
            }

        description = (data.get("Description") or data.get("description") or "") or ""
        status = (data.get("Status") or data.get("status") or "To Do") or "To Do"

        with _conn() as conn:
            with conn.cursor() as cur:
                cur.execute(
                    """
                    INSERT INTO Tasks (Title, Description, Status)
                    VALUES (%s, %s, %s)
                    """,
                    (title, description, status),
                )
                new_id = cur.lastrowid

        created = {
            "Id": int(new_id),
            "Title": title,
            "Description": description,
            "Status": status,
        }
        return {"statusCode": 201, "headers": _headers(), "body": json.dumps(created)}

    except Exception as e:
        # Optional: print for CloudWatch logs
        print(f"ERROR: {e}")
        return {
            "statusCode": 500,
            "headers": _headers(),
            "body": json.dumps({"message": "Internal Server Error"}),
        }


